File: app/src/main/java/com/soutvoid/gamesproject/ui/screen/personalize/PersonalizeActivityPresenter.java
package com.soutvoid.gamesproject.ui.screen.personalize;

import com.agna.ferro.mvp.component.scope.PerScreen;
import com.soutvoid.gamesproject.interactor.util.ExploreQuery;
import com.soutvoid.gamesproject.ui.base.activity.BasePresenter;
import com.soutvoid.gamesproject.ui.common.error.ErrorHandler;

import javax.inject.Inject;

import io.realm.Realm;
import io.realm.RealmResults;

@PerScreen
public class PersonalizeActivityPresenter extends BasePresenter<PersonalizeActivityView> {

    private Realm realm;

    @Inject
    public PersonalizeActivityPresenter(ErrorHandler errorHandler) {
        super(errorHandler);
    }

    @Override
    public void onLoad(boolean viewRecreated) {
        super.onLoad(viewRecreated);

    }

    @Override
    public void onResume() {
        super.onResume();
        refreshQueries();
    }

    private synchronized void refreshQueries() {
        realm = Realm.getDefaultInstance();
        RealmResults<ExploreQuery> exploreQueries = realm.where(ExploreQuery.class).findAll();
        getView().setQueriesContent(realm.copyFromRealm(exploreQueries));
        realm.close();
    }
}

File: app/src/main/java/com/soutvoid/gamesproject/ui/screen/personalize/PersonalizeActivityPresenter.kt
package com.soutvoid.gamesproject.ui.screen.personalize

import com.agna.ferro.mvp.component.scope.PerScreen
import com.soutvoid.gamesproject.interactor.util.ExploreQuery
import com.soutvoid.gamesproject.ui.base.activity.BasePresenter
import com.soutvoid.gamesproject.ui.common.error.ErrorHandler
import io.realm.Realm
import javax.inject.Inject

@PerScreen
class PersonalizeActivityPresenter @Inject
constructor(errorHandler: ErrorHandler) : BasePresenter<PersonalizeActivityView>(errorHandler) {

    private var realm: Realm? = null

    override fun onLoad(viewRecreated: Boolean) {
        super.onLoad(viewRecreated)

    }

    override fun onResume() {
        super.onResume()
        refreshQueries()
    }

    @Synchronized private fun refreshQueries() {
        realm = Realm.getDefaultInstance()
        val exploreQueries = realm!!.where(ExploreQuery::class.java).findAll()
        view.setQueriesContent(realm!!.copyFromRealm(exploreQueries))
        realm!!.close()
    }
}
